.github/workflows/push-realease-candidate-tag.yml

```diff
@@ -0,0 +1,72 @@
+name: 'Push RC Tag After Merge'
+
+on:
+  push:
+    branches:
+      - 'releases/**'
+
+permissions:
+  contents: write  # Required to push tags using the GITHUB_TOKEN
+
+jobs:
+  tag-rc-version:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Full history required to create tags
+          token: ${{ secrets.RELEASE_BOT_TOKEN }}
+
+      - name: Extract RC tag from commit message
+        id: extract_tag
+        run: |
+          COMMIT_MSG="${{ github.event.head_commit.message }}"
+          echo "Commit message: $COMMIT_MSG"
+
+          # Extract the tag version from the message (e.g., v1.0.214-rc3)
+          RC_TAG=$(echo "$COMMIT_MSG" | grep -oE '^This is an automated PR for the release candidate v[0-9]+\.[0-9]+\.[0-9]+-rc[0-9]+' | grep -oE 'v[0-9]+\.[0-9]+\.[0-9]+-rc[0-9]+')
+
+          if [[ -z "$RC_TAG" ]]; then
+            echo "No RC tag found in commit message. Skipping."
+            exit 0
+          fi
```

Comment:
so the way this works is that the commit message has to have an `rc` version in it for the new tag to be added?

Looking forward, we may want to drop multiple commits into a release branch before releasing, for example if we start doing patch version releases and backports for bugfixes to older versions. We'll probably want to think about this pattern and how to prevent someone from accidentally publishing a new rc release based on their commit message